genomes/models/genome.py

```diff
@@ -0,0 +1,226 @@
+    num_proteins = models.IntegerField(db_column="num_proteins")
+    eggnog_coverage = models.FloatField(db_column="eggnog_coverage")
+    ipr_coverage = models.FloatField(db_column="ipr_coverage")
+    taxon_lineage = models.CharField(db_column="taxon_lineage", max_length=400)
+
+    num_genomes_total = models.IntegerField(
+        db_column="pangenome_total_genomes", null=True, blank=True
+    )
+    pangenome_size = models.IntegerField(
+        db_column="pangenome_size", null=True, blank=True
+    )
+    pangenome_core_size = models.IntegerField(
+        db_column="pangenome_core_prop", null=True, blank=True
+    )
+    pangenome_accessory_size = models.IntegerField(
+        db_column="pangenome_accessory_prop", null=True, blank=True
+    )
+
+    annotations = models.JSONField(default=default_annotations, db_column="annotations")
+    default_annotations = staticmethod(default_annotations)
+
+    last_update = models.DateTimeField(db_column="last_update", auto_now=True)
+    first_created = models.DateTimeField(db_column="first_created", auto_now_add=True)
+
+    geo_origin = models.ForeignKey(
+        "GeographicLocation",
+        db_column="geographic_origin",
+        null=True,
+        blank=True,
+        on_delete=models.PROTECT,
+    )
+
+    pangenome_geographic_range = models.ManyToManyField(
+        "GeographicLocation",
+        db_table="genome_pangenome_geographic_range",
+        related_name="geographic_range",
+    )
+
+    result_directory = models.CharField(
+        db_column="result_directory", max_length=100, blank=True, null=True
+    )
+
+    catalogue = models.ForeignKey(
+        "GenomeCatalogue",
+        db_column="genome_catalogue",
+        on_delete=models.PROTECT,
+        related_name="genomes",
+    )
+
+    @property
+    def geographic_range(self):
+        return [v.name for v in self.pangenome_geographic_range.all()]
+
+    @property
+    def geographic_origin(self):
+        if self.geo_origin:
+            name = self.geo_origin.name
+        else:
+            name = None
+        return name
+
+    @property
+    def last_update_iso(self):
+        return self.last_update.isoformat() if self.last_update else None
+
+    @property
+    def first_created_iso(self):
+        return self.first_created.isoformat() if self.first_created else None
+
+    def __str__(self):
+        return self.accession
+
+
+class GenomeAssemblyLink(models.Model):
+    """
+    A many-to-many relationship model between Genome and Assembly.
+    Stores additional information about the relationship.
+    """
+
+    genome = models.ForeignKey(
+        Genome, on_delete=models.CASCADE, related_name="assembly_links"
+    )
+    assembly = models.ForeignKey(
+        Assembly, on_delete=models.CASCADE, related_name="genome_links"
+    )
+    species_rep = models.CharField(
+        max_length=100,
+        null=True,
+        blank=True,
+        help_text="Arbitrary genome accession for species representative",
+    )
```

Comment:
Shouldn't this one be a "Genome" too?

Reply:
Not clear on what you mean here. 
But species_rep was added as it's needed for MAG assembly linkage